libbeat/tests/integration/framework.go

```diff
@@ -294,6 +357,83 @@ func createTempDir(t *testing.T) string {
 // using the default test credentials or the corresponding environment
 // variables.
 func EnsureESIsRunning(t *testing.T) {
+	esURL := GetESURL(t, "http")
+
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(500*time.Second))
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, esURL.String(), nil)
+	if err != nil {
+		t.Fatalf("cannot create request to ensure ES is running: %s", err)
+	}
+
+	u := esURL.User.Username()
+	p, _ := esURL.User.Password()
+	req.SetBasicAuth(u, p)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		// If you're reading this message, you probably forgot to start ES
+		// run `mage compose:Up` from Filebeat's folder to start all
+		// containers required for integration tests
+		t.Fatalf("cannot execute HTTP request to ES: %s", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("unexpected HTTP status: %d, expecting 200 - OK", resp.StatusCode)
+	}
+}
+
+func (b *BeatProc) FileContains(filename string, match string) string {
+	file, err := os.Open(filename)
+	require.NoErrorf(b.t, err, "error opening: %s", filename)
+	r := bufio.NewReader(file)
+	for {
+		line, err := r.ReadString('\n')
+		if err != nil {
+			if err != io.EOF {
+				b.t.Fatalf("error reading log file '%s': %s", file.Name(), err)
+			}
+			break
+		}
+		if strings.Contains(line, match) {
+			return line
+		}
+	}
+	return ""
+}
+
+func (b *BeatProc) WaitFileContains(filename string, match string, waitFor time.Duration) string {
+	var returnValue string
+	require.Eventuallyf(b.t,
+		func() bool {
+			returnValue = b.FileContains(filename, match)
+			return returnValue != ""
+		}, waitFor, 100*time.Millisecond, "match string '%s' not found in %s", match, filename)
+
+	return returnValue
+}
+
+func (b *BeatProc) WaitStdErrContains(match string, waitFor time.Duration) string {
+	return b.WaitFileContains(b.stderr.Name(), match, waitFor)
+}
+
+func (b *BeatProc) WaitStdOutContains(match string, waitFor time.Duration) string {
+	return b.WaitFileContains(b.stdout.Name(), match, waitFor)
+}
+
+func (b *BeatProc) LoadMeta() (Meta, error) {
+	m := Meta{}
+	metaFile, err := os.Open(filepath.Join(b.TempDir(), "data", "meta.json"))
+	if err != nil {
+		return m, err
+	}
+	defer metaFile.Close()
+
+	metaBytes, _ := ioutil.ReadAll(metaFile)
+	json.Unmarshal(metaBytes, &m)
+	return m, nil
```

Comment:
[Blocker]
Why are you ignoring the errors? If either of them aren't `nil` the returned value cannot be used.

Reply:
🤦 fixed